sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -284,7 +307,9 @@ class FrameLessOffsetWindowFunctionFrame(
       }
   } else {
     (current: InternalRow) =>
-      if (inputIndex >= 0 && inputIndex < input.length) {
+      if (absOffset > input.length) {
```

Comment:
Let's avoid changing the write path.

Reply:
This refactor need the change.
Before this refactor, `inputIndex = offset` if `offset > rows.length`. 
After this refactor, `inputIndex = 0` if `offset > rows.length`.

Reply:
Then let's restore this behavior

Reply:
should the entire write path do nothing if `absOffset > input.length`?

Reply:
Good idea.